src/teachy_tv.c

```diff
@@ -0,0 +1,1775 @@
+{
+    int hasItem;
+    gMultiuseListMenuTemplate = gUnknown_8479368;
+    gMultiuseListMenuTemplate.windowId = 1;
+    gMultiuseListMenuTemplate.moveCursorFunc = TeachyTvAudioByInput;
+    hasItem = (u8)CheckBagHasItem(ITEM_TM_CASE, 1);
+    if ( !(hasItem << 24) )
+    {
+        gMultiuseListMenuTemplate.items = &gUnknown_8479340;
+        gMultiuseListMenuTemplate.totalItems = 5;
+        gMultiuseListMenuTemplate.maxShowed = 5;
+        gMultiuseListMenuTemplate.upText_Y = (gMultiuseListMenuTemplate.upText_Y + 8) & 0xF;
+    }
+    return ListMenuInit(
+               &gMultiuseListMenuTemplate,
+               gTeachyTV_StaticResources.scrollOffset,
+               gTeachyTV_StaticResources.selectedRow);
+}
+
+void TeachyTvSetupScrollIndicatorArrowPair()
+{
+    int hasItem;
+    hasItem = (u8)CheckBagHasItem(ITEM_TM_CASE, 1);
+    if ( ! hasItem << 24 )
+    {
+        struct TeachyTvBuf * temp = gUnknown_203F450;
+        temp->var_4007 = 0xFF;
+    }
+
+    else {
+        struct TeachyTvBuf *temp; 
+        u8 res;
+        res = AddScrollIndicatorArrowPair(&gUnknown_8479380, &(gTeachyTV_StaticResources.scrollOffset));
+        temp = gUnknown_203F450;
+        temp->var_4007 = res;
+        }
+}
+
+void TeachyTvRemoveScrollIndicatorArrowPair()
+{
+    u8 *temp = &(gUnknown_203F450->var_4007);
+    if ( *temp != 0xFF )
+    {
+        RemoveScrollIndicatorArrowPair(*temp);
+        gUnknown_203F450->var_4007 = 0xFF;
+    }
+}
+
+void TeachyTvAudioByInput(s32 notUsed, bool8 play, struct ListMenu *notUsedAlt)
+{
+    if ( play != 1 )
+        PlaySE(SE_SELECT);
+}
+
+void TeachyTvInitIo()
+{
+    SetGpuReg(REG_OFFSET_WININ, 0x3F);
+    SetGpuReg(REG_OFFSET_WINOUT, 0x1F);
+    SetGpuReg(REG_OFFSET_BLDCNT, 0xCC);
+    SetGpuReg(REG_OFFSET_BLDY, 0x5);
+}
+
+u8 TeachyTvSetupObjEventAndOam()
+{
+    u8 temp = AddPseudoEventObject(90, SpriteCallbackDummy, 0, 0, 8);
+    gSprites[temp].oam.priority = 2;
+    gSprites[temp].invisible = 1;
+    return temp;
+}
+
+void TeachyTvSetSpriteCoordsAndSwitchFrame(u8 objId, u16 x, u16 y, u8 frame)
+{
+    gSprites[objId].pos2.x = x;
+    gSprites[objId].pos2.y = y;
+    gSprites[objId].invisible = 0;
+    StartSpriteAnim(&gSprites[objId], frame);
+}
+
+void TeachyTvSetWindowRegs()
+{
+    SetGpuReg(REG_OFFSET_WIN0V, 0xC64);
+    SetGpuReg(REG_OFFSET_WIN0H, 0x1CD4);
+}
+
+void TeachyTvClearWindowRegs()
+{
+    SetGpuReg(REG_OFFSET_WIN0V, 0x0);
+    SetGpuReg(REG_OFFSET_WIN0H, 0x0);
+}
+
+void TeachyTvBg2AnimController()
+{
+    u16 *tilemapBuffer;
+    u8 i;
+    u32 offset, offset2, j;
+
+    tilemapBuffer = (u16 *)GetBgTilemapBuffer(2);
+    i = 1;
+    do
+    {
+        offset2 = 2;
+        offset = 0x20 * i;
+        j = i + 1;
+        do
+        {
+            tilemapBuffer[offset + offset2] = ((Random() & 3) << 10) + 0x301F;
+            offset2 = (offset2 + 1) << 0x18 >> 0x18;
+        }
+        while ( offset2 <= 0x1B );
+        i = j;
+    }
+    while ( j << 0x18 >> 0x18 <= 0xC );
```

Comment:
`j` is a `u8` variable

Reply:
I improved a bit, but if j is a u8 var it will cause additional shift at `j = i + 1;`